### DISCUSSION TITLE
Load feature flags

### DISCUSSION THREAD

**Discussion by guihermegambeti** [statelyai/xstate#3716] [General]
Hi there, sorry to disturb you with a question.

I've tried all the potential solutions that came to mind to address this use-case, however, I didn't find a satisfactory one.

**What we need to do**: We need to feature flag a few things in our current state machine
**My ideal solution:** Always load, no matter what state is, all feature flags and assign them to the state machine context
**Attempts:** Tried using async actions and invoke services, however, I cannot find a way to always run either of them

Does anyone have any idea of how to implement such use case?

**Comment by davidkpiano** [comment]
Can you share the current code that you have trying this? Are these async feature flags? And what is meant by "always run"?

**Reply by guihermegambeti** [reply]
Thank you @davidkpiano for getting some time to help us.

Yes, there are async feature flags.
By "always run" I mean that I'm looking for a way to always load feature flags into the state machine context before entering a given state, no matter what the state is, so that I can drive the state transition based on the feature flags in the context.

This basically my state machine and how I envisioned loading feature flag. However, the `invoke.src` function just gets called for the first time when I'm first loading the state machine. 
Every time that I hydrate the state machine and the state machine is in one of the state, for example create, the `invoke.src` function does not get called therefore no FF is loaded into the context

```
const stateMachine = createStateMachine({
    id: 'state-machine',
    invoke: {
        src: async () => {
           return await featureFlagService.load();
        },
       onDone: {
         actions: assign(_context, event) => ({ featureFlagEvaluations: event.data }),
       }
    },
    states: {
        'create': { ... },
        'retrieve': { ... },
    }
});
```

I hope this clarifies a little bit
